crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -1458,36 +1458,44 @@ mod tests {
     };
     use reth_execution_types::{Chain, ExecutionOutcome};
     use reth_primitives::{
-        BlockHashOrNumber, BlockNumHash, BlockNumberOrTag, BlockWithSenders, Header, Receipt,
-        SealedBlock, SealedBlockWithSenders, StaticFileSegment, Withdrawals, B256,
+        BlockHashOrNumber, BlockNumHash, BlockNumberOrTag, BlockWithSenders, Receipt, SealedBlock,
+        SealedBlockWithSenders, StaticFileSegment, TransactionMeta, TransactionSignedNoHash,
+        Withdrawals, B256,
     };
     use reth_storage_api::{
         BlockHashReader, BlockIdReader, BlockNumReader, BlockReader, BlockReaderIdExt, BlockSource,
         ChangeSetReader, HeaderProvider, ReceiptProviderIdExt, RequestsProvider,
-        TransactionVariant, WithdrawalsProvider,
+        TransactionVariant, TransactionsProvider, WithdrawalsProvider,
     };
     use reth_testing_utils::generators::{
         self, random_block, random_block_range, random_changeset_range, random_eoa_accounts,
-        random_receipt, random_signed_tx, BlockParams, BlockRangeParams,
+        random_receipt, BlockParams, BlockRangeParams,
     };
     use revm::db::BundleState;
 
     const TEST_BLOCKS_COUNT: usize = 5;
 
+    const TEST_TRANSACTIONS_COUNT: u8 = 4;
+
     fn random_blocks(
         rng: &mut impl Rng,
         database_blocks: usize,
         in_memory_blocks: usize,
         requests_count: Option<Range<u8>>,
         withdrawals_count: Option<Range<u8>>,
+        tx_count: Range<u8>,
     ) -> (Vec<SealedBlock>, Vec<SealedBlock>) {
         let block_range = (database_blocks + in_memory_blocks - 1) as u64;
         let blocks = random_block_range(
             rng,
             0..=block_range,
             BlockRangeParams {
                 parent: Some(B256::ZERO),
-                tx_count: 0..1,
+                tx_count: if tx_count.end == tx_count.start {
+                    tx_count.start..tx_count.start + 1
+                } else {
+                    tx_count
+                },
```

Comment:
nit: you can accept `impl RangeBounds<u8>`, and convert into `Range` here, so you can explicitly pass `TEST_TRANSACTIONS_COUNT..=TEST_TRANSACTIONS_COUNT`

Reply:
I'm not sure if this solves the problem... Indeed this if/else is there for the case where `BlockRangeParams` is default (for about half of the tests) and so the range is 0..0 which is not accepted by the `random_block_range` method which requires at least `0..1` (which was hardcoded before). That's why I did this trick, I think it's enough right?

Reply:
if you pass `0..=0` to this method, it should get converted to `0..1` which is basically the same as it was before, but more explicit

Reply:
Just fixed, let me know if this is what you had in mind here